subsys/disk/disk_access.c

```diff
@@ -58,9 +58,19 @@ int disk_access_init(const char *pdrv)
 	struct disk_info *disk = disk_access_get_di(pdrv);
 	int rc = -EINVAL;
 
-	if ((disk != NULL) && (disk->ops != NULL) &&
-				(disk->ops->init != NULL)) {
-		rc = disk->ops->init(disk);
+	if ((disk != NULL) && (disk->refcnt == 0U)) {
+		/* Disk has not been initialized, start it */
+		if ((disk->ops != NULL) && (disk->ops->init != NULL)) {
+			rc = disk->ops->init(disk);
+			if (rc == 0) {
+				/* Increment reference count */
+				disk->refcnt++;
+			}
+		}
+	} else if ((disk != NULL) && (disk->refcnt < UINT16_MAX)) {
```

Comment:
I know that possibility that refcnt gets to `UINT16_MAX` is impossible, but what is the point of just stopping increment when we get to `disk->refcnt` without message? I think that when we have reached that point there should be error issued, because something seems to call init for no reason in loop.

Reply:
We would issue an error, right? both the `if` and `else if` cases would be false, so we would return `-EINVAL`. Or do you think we should also add a logging statement

Reply:
I mean logging. Does this make sense?

Reply:
Sure, I have no issue adding that